tests/integration-tests-utils/pom.xml

```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="
+  http://maven.apache.org/POM/4.0.0
+  http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.bookkeeper.tests</groupId>
+    <artifactId>tests-parent</artifactId>
+    <version>4.7.0-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.apache.bookkeeper.tests</groupId>
+  <artifactId>integration-tests-utils</artifactId>
+  <packaging>jar</packaging>
+
+  <name>Apache BookKeeper :: Tests :: Utility module for Arquillian based integration tests</name>
+
+  <properties>
+    <shrinkwrap.version>3.0.1</shrinkwrap.version>
+    <arquillian-cube.version>1.12.1-SNAPSHOT</arquillian-cube.version>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-compress</artifactId>
+      <version>1.15</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.jboss.shrinkwrap.resolver</groupId>
+      <artifactId>shrinkwrap-resolver-impl-maven</artifactId>
+      <version>${shrinkwrap.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jboss.shrinkwrap.resolver</groupId>
+      <artifactId>shrinkwrap-resolver-api</artifactId>
+      <version>${shrinkwrap.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.zookeeper</groupId>
```

Comment:
I would like to point out that having zookeeper jar on the main classloader will make every client use "this" version of ZK client (or a mix of this an of the one downloaded from Maven).

This may not a problem, because ZK client is quite stable. So for me we can keep this setup if it makes things simpler.
We are already setting up a complex environment, I don't know it if it worth to load ZooKeeper client in BookKeeperClusterUtils.java using our brand new MavenClassLoader and drop this dependency at all here

Reply:
I ran into this from another angle earlier. It can be solved with a isolated classloader for the surefire test, and passing the systemclassloader as the parent classloader. messes up log4j though, so looking into it more.

Reply:
Latest patch solves this. Tests need to run with forkCount=0 though to get isolated classloader (the other methods for doing so don't seem to work).

Reply:
great!
what about adding a test that certifies that a ZooKeeper handle created using a MavenClassLoader is not loaded from the boot classloader ?

another issue: please check tests with Java9 too, the hierarchy of main classloaders is not "changed" but the "boot" classloader now does not load every class in the JDK, but only a minimal subset (like java.lang, java.io....)

Reply:
Added test for zookeeper (for KeeperException, because having to create a watcher would suck).

Jenkins should flag any java9 issue, but I wouldn't expect any. system != boot. system is the classloader that apps usually use. The isolated classloaders are something that surefire does (http://maven.apache.org/surefire/maven-surefire-plugin/examples/class-loading.html).

Reply:
sounds good to me, thank you